satellite/gracefulexit/endpoint.go

```diff
@@ -620,3 +633,39 @@ func (endpoint *Endpoint) handleFailed(ctx context.Context, pending *pendingMap,
 
 	return nil
 }
+
+func (endpoint *Endpoint) getFinishedMessage(ctx context.Context, signer signing.Signer, nodeID storj.NodeID, finishedAt time.Time, success bool, reason pb.ExitFailed_Reason) (message *pb.SatelliteMessage, err error) {
+	if success {
+		unsigned := &pb.ExitCompleted{
+			SatelliteId: endpoint.signer.ID(),
+			NodeId:      nodeID,
+			Completed:   finishedAt,
+		}
+		signed, err := signing.SignExitCompleted(ctx, endpoint.signer, unsigned)
+		if err != nil {
+			return nil, Error.Wrap(err)
+		}
+		message = &pb.SatelliteMessage{Message: &pb.SatelliteMessage_ExitCompleted{
+			ExitCompleted: signed,
+		}}
+	} else {
+		unsigned := &pb.ExitFailed{
+			SatelliteId: endpoint.signer.ID(),
+			NodeId:      nodeID,
+			Failed:      finishedAt,
+			Reason:      reason,
+		}
+		if reason >= 0 {
```

Comment:
Is this necessary? On line 656, we are already assign `reason` to `unsigned.Reason` right?

Reply:
Maybe we can change this check to the beginning of this `else` block since we probably should have a `reason` whenever we send a `ExitFailed` message

Reply:
Oops.  Removing 656 so that we check for valid reason.

Reply:
In regards to:
Maybe we can change this check to the beginning of this else block since we probably should have a reason whenever we send a ExitFailed message

It's possible the SN connects again and we need to resend the failure.  I had a comment that maybe we need to store the reason in the DB so we can recreate the failure signature. Thoughts?

Reply:
I'm not sure about this one either-- do you think it would take up a lot of space in the db? I guess what are the cons of not saving the original reason/being able to recreate the appropriate failure signature?